cdap-security-spi/src/test/java/co/cask/cdap/security/spi/authorization/AuthorizerTest.java

```diff
@@ -115,6 +117,100 @@ public void testHierarchy() throws UnauthorizedException {
     verifyAuthFailure(namespace, user, Action.READ);
   }
 
+  @Test
+  public void testRBAC() throws Exception {
+    Authorizer authorizer = get();
+
+    Role admins = new Role("admins");
+    Role engineers = new Role("engineers");
+    // create a role
+    authorizer.createRole(admins);
+    // add another role
+    authorizer.createRole(engineers);
+
+    // listing role should show the added role
+    Set<Role> roles = authorizer.listAllRoles();
+    Assert.assertEquals(new HashSet<>(Arrays.asList(admins, engineers)), roles);
```

Comment:
for consistency: ``Collections.unmodifiableSet(Arrays.asList(admins, engineers))``